content_scripts/storageLoader.js

```diff
@@ -195,6 +196,7 @@ async function loadSave() {
         setLineSpacing(parsedData.EnhancePage.lineSpacing)
         // CourseRemover
         removeCoursesByConfiguration(parsedData);
+        resetCourseRemoverStatus(parsedData);
```

Comment:
Every time you enter the site, the extension should start with the course remover state on "OFF". 
Otherwise, you can leave the site with "ON" mode and then on your next entrance the button will be on "ON" mode while the course remover is closed (causing the state to be reversed all the time)

Reply:
I understand I think that the course removed functionality needs to be aware of it no?

Reply:
I'm nor sure, the state should save only if the user pressed the btn and it's on or not. 
There is no way to know that since the script that adds the "X" is running inside the tab, and uses a global variable that is not reachable from the popup.js
so it had to be a different mechanism. @norbit8 is there a way to know if it's open?

Reply:
I think that as the courseRemover.js can access functions on the storageLoader you can manipulate it to load the localstorage and update it accordingly

Reply:
I don't think that the courseRemover script should be aware of that.
The only "bug" I can think of is that if you click on the course remover button and then refresh, I think that the 'x' buttons still appear because `window.ANYTHING` is a global variable which is not resets when the page gets refreshed.

So, we can either ignore it or try to fix that.
IMO we should override thw global var on every refresh in the courseRemover script so it will comply with this PR functionality.
(I can fix that in a new PR and issue if you let me 😁)

Reply:
> The only "bug" I can think of is that if you click on the course remover button and then hit refresh, I think that the 'x' buttons still appear because `window.ANYTHING` is a global variable which does not reset when the page gets refreshed.
> 
> So, we can either ignore it or try to fix that.
> IMO we should override the global var on every refresh in the courseRemover script so it will comply with this PR functionality.
> (I can fix that in a new PR and issue if you let me 😁)

Yeah, I think it's better to use the localstorage data instead of the global var.
Regarding that you fix it in new PR I'm fine with it :) as long as Yanir is

Maybe when you fix it create some function in the localStorage script that will return the MoodleBooster settings so other scripts will be able to use it.

Reply:
> > The only "bug" I can think of is that if you click on the course remover button and then hit refresh, I think that the 'x' buttons still appear because `window.ANYTHING` is a global variable which does not reset when the page gets refreshed.
> > 
> > So, we can either ignore it or try to fix that.
> > IMO we should override the global var on every refresh in the courseRemover script so it will comply with this PR functionality.
> > (I can fix that in a new PR and issue if you let me 😁)
> 
> Yeah, I think it's better to use the localstorage data instead of the global var.
> Regarding that you fix it in new PR I'm fine with it :) as long as Yanir is
> 
> Maybe when you fix it create some function in the localStorage script that will return the MoodleBooster settings so other scripts will be able to use it.
> 
> 

I think yanir created this function, right @yanirelfasy?

Reply:
For now, I'm merging the PR.
If the feature doesn't exist I'll be happy if you'll add it so everything will be more generic